design/logging.md

```diff
@@ -0,0 +1,495 @@
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/cookies.rb:670:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
+/var/lib/gems/2.5.0/gems/activesupport-5.2.4.2/lib/active_support/callbacks.rb:98:in `run_callbacks'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/debug_exceptions.rb:61:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
+/var/lib/gems/2.5.0/gems/railties-5.2.4.2/lib/rails/rack/logger.rb:38:in `call_app'
+/var/lib/gems/2.5.0/gems/railties-5.2.4.2/lib/rails/rack/logger.rb:26:in `block in call'
+/var/lib/gems/2.5.0/gems/activesupport-5.2.4.2/lib/active_support/tagged_logging.rb:71:in `block in tagged'
+/var/lib/gems/2.5.0/gems/activesupport-5.2.4.2/lib/active_support/tagged_logging.rb:28:in `tagged'
+/var/lib/gems/2.5.0/gems/activesupport-5.2.4.2/lib/active_support/tagged_logging.rb:71:in `tagged'
+/var/lib/gems/2.5.0/gems/railties-5.2.4.2/lib/rails/rack/logger.rb:26:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/request_id.rb:27:in `call'
+/var/lib/gems/2.5.0/gems/rack-2.2.2/lib/rack/method_override.rb:24:in `call'
+/var/lib/gems/2.5.0/gems/rack-2.2.2/lib/rack/runtime.rb:22:in `call'
+/var/lib/gems/2.5.0/gems/activesupport-5.2.4.2/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/executor.rb:14:in `call'
+/var/lib/gems/2.5.0/gems/actionpack-5.2.4.2/lib/action_dispatch/middleware/static.rb:127:in `call'
+/var/lib/gems/2.5.0/gems/rack-2.2.2/lib/rack/sendfile.rb:110:in `call'
+/var/lib/gems/2.5.0/gems/conjur-rack-heartbeat-2.2.0/lib/rack/heartbeat.rb:20:in `call'
+/var/lib/gems/2.5.0/gems/railties-5.2.4.2/lib/rails/engine.rb:524:in `call'
+/var/lib/gems/2.5.0/gems/puma-3.12.4/lib/puma/configuration.rb:227:in `call'
+/var/lib/gems/2.5.0/gems/puma-3.12.4/lib/puma/server.rb:675:in `handle_request'
+/var/lib/gems/2.5.0/gems/puma-3.12.4/lib/puma/server.rb:476:in `process_client'
+/var/lib/gems/2.5.0/gems/puma-3.12.4/lib/puma/server.rb:334:in `block in run'
+/var/lib/gems/2.5.0/gems/puma-3.12.4/lib/puma/thread_pool.rb:135:in `block in spawn_thread'
+ INFO 2020/05/17 17:38:07 +0000 [pid=300] [origin=127.0.0.1] [request_id=d3f54689-ed9d-4c66-858a-d01d9c7adbc9] [tid=317] Completed 401 Unauthorized in 447ms
+```
+
+Here we can see that the log is more robust and has all the data that can help a developer
+to investigate the issue, including the stack trace and the DB messages.
+
+### Backwards compatibility
+
+It's important to say that the UX will not change by default. We will do this by changing
```

Comment:
While this doesn't change for new setups, it will change for users that have a specific value set for `CONJUR_LOG_LEVEL`. I think it's worth mentioning that when not using the default value, there will be an expected behavior change.